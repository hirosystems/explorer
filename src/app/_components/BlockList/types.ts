import { Block } from '@stacks/stacks-blockchain-api-types';

export type EnhancedBlock = Block & { destroy?: boolean; animate?: boolean };

export enum UIBlockType {
  Block = 'block',
  BurnBlock = 'burn_block',
  Count = 'count',
}

export interface UISingleBlock {
  type: UIBlockType.Block | UIBlockType.BurnBlock;
  height: number | string;
  hash: string;
  timestamp: number;
  txsCount?: number;
}

export interface UIBlockCount {
  type: UIBlockType.Count;
  count: number;
}

export type UIBlock = UISingleBlock | UIBlockCount;
