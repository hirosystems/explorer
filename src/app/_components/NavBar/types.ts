import { ReactNode } from 'react';

export interface NavItem {
  id: string;
  label: string | ReactNode;
  children?: NavItem[];
  href?: string;
  onClick?: () => void;
}
