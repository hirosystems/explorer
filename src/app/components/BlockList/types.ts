import { Block } from '@stacks/blockchain-api-client';

export type EnhancedBlock = Block & { destroy?: boolean; animate?: boolean };
