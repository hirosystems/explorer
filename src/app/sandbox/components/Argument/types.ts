import React from 'react';

import { ClarityAbiType } from '@stacks/transactions';

import { ListValueType, ValueType } from '../../types/values';

export interface ArgumentInputProps {
  type: ClarityAbiType;
  value: ValueType | ListValueType;
}

export interface CommonArgumentInputProps {
  name: string;
  handleChange: (e: React.FormEvent<HTMLInputElement>) => void;
  error: any;
}
