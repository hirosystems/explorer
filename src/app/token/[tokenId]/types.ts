export interface TokenLinks {
  websites: string[];
  blockchain: string[];
  chat: string[];
  forums: string[];
  announcements: string[];
  repos: string[];
  social: string[];
}

export interface DeveloperData {
  forks: number | null;
  stars: number | null;
  subscribers: number | null;
  total_issues: number | null;
  closed_issues: number | null;
  pull_requests_merged: number | null;
  pull_request_contributors: number | null;
  code_additions_deletions_4_weeks: {
    additions: number | null;
    deletions: number | null;
  } | null;
  commit_count_4_weeks: number | null;
  last_4_weeks_commit_activity_series: [] | null;
}

export interface BasicTokenInfo {
  name: string;
  symbol: string;
  totalSupply: number | null;
}

export interface TokenInfoProps {
  basic?: BasicTokenInfo;
  extended?: {
    categories: string[];

    links: TokenLinks;
    circulatingSupply: number | null;
    fullyDilutedValuation: number | null;
    tvl: number | null;

    currentPrice: number | null;
    priceChangePercentage24h: number | null;
    currentPriceInBtc: number | null;
    priceInBtcChangePercentage24h: number | null;

    marketCap: number | null;

    tradingVolume24h: number | null;
    tradingVolumeChangePercentage24h: number | null;

    developerData: DeveloperData;
  };
}
