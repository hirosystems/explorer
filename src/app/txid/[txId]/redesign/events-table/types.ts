export enum EventsTableColumns {
  Index = 'index',
  AssetEventType = 'asset-event-type',
  Asset = 'asset',
  EventType = 'event-type',
  Amount = 'amount',
  From = 'from',
  ArrowRight = 'arrow-right',
  To = 'to',
}
