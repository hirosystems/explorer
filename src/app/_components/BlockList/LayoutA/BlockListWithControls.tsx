import React from 'react';

import { ListFooter } from '../../../../common/components/ListFooter';
import { Section } from '../../../../common/components/Section';
import { Box } from '../../../../ui/Box';
import { Controls } from '../Controls';
import { UIBlock } from '../types';
import { Blocks } from './Blocks';
import { UpdateBar } from './UpdateBar';
import { useBlockListContext } from './context';

export function BlockListWithControls({
  blockList,
  latestBlocksCount,
  updateList,
  enablePagination,
  isFetchingNextPage,
  hasNextPage,
  fetchNextPage,
  horizontalControls,
}: {
  blockList: UIBlock[];
  latestBlocksCount: number;
  updateList: () => void;
  enablePagination?: boolean;
  isFetchingNextPage?: boolean;
  hasNextPage?: boolean;
  fetchNextPage?: () => void;
  horizontalControls?: boolean;
}) {
  const {
    isUpdateListLoading,
    setIsUpdateListLoading,
    groupedByBtc,
    setGroupedByBtc,
    liveUpdates,
    setLiveUpdates,
  } = useBlockListContext();
  return (
    <Section title="Recent Blocks">
      <Box pb={6}>
        <Controls
          groupByBtc={{
            onChange: () => {
              setGroupedByBtc(!groupedByBtc);
            },
            isChecked: groupedByBtc,
            isDisabled: true,
          }}
          liveUpdates={{
            onChange: () => setLiveUpdates(!liveUpdates),
            isChecked: liveUpdates,
          }}
          horizontal={horizontalControls}
        />
        {!liveUpdates && (
          <UpdateBar
            isUpdateListLoading={isUpdateListLoading}
            latestBlocksCount={latestBlocksCount}
            onClick={updateList}
          />
        )}
        <Blocks blockList={blockList} isUpdateListLoading={isUpdateListLoading} />
        <Box pt={4}>
          {(!liveUpdates || !enablePagination) && (
            <ListFooter
              isLoading={isFetchingNextPage}
              hasNextPage={hasNextPage}
              fetchNextPage={fetchNextPage}
              label={'blocks'}
            />
          )}
        </Box>
      </Box>
    </Section>
  );
}
