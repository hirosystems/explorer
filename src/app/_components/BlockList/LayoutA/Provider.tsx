import { ReactNode, useState } from 'react';

import { BlockListContext } from './context';

export function BlockListProvider({ children }: { children: ReactNode }) {
  const [isUpdateListLoading, setIsUpdateListLoading] = useState(false);
  const [groupedByBtc, setGroupedByBtc] = useState(true);
  const [liveUpdates, setLiveUpdates] = useState(false);

  return (
    <BlockListContext.Provider
      value={{
        isUpdateListLoading,
        setIsUpdateListLoading,
        groupedByBtc,
        setGroupedByBtc,
        liveUpdates,
        setLiveUpdates,
      }}
    >
      {children}
    </BlockListContext.Provider>
  );
}
