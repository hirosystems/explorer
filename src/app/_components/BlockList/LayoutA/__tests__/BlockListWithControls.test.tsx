import { render } from '@testing-library/react';

import { UIBlock, UIBlockType } from '../../types';
import { BlockListWithControls } from '../BlockListWithControls';
import { BlockListProvider } from '../Provider';

const date = new Date('2024-01-01T00:00:00Z').getTime();

describe('BlockListWithControls', () => {
  it('renders correctly', () => {
    const blockList: UIBlock[] = [
      {
        type: UIBlockType.Block,
        height: 1,
        hash: 'hash1',
        timestamp: date,
        txsCount: 5,
      },
      {
        type: UIBlockType.Block,
        height: 2,
        hash: 'hash2',
        timestamp: date,
        txsCount: 10,
      },
      {
        type: UIBlockType.Block,
        height: 3,
        hash: 'hash3',
        timestamp: date,
        txsCount: 15,
      },
      {
        type: UIBlockType.BurnBlock,
        height: 4,
        hash: 'hash4',
        timestamp: date,
        txsCount: 30,
      },
      {
        type: UIBlockType.Count,
        count: 25,
      },
    ];
    const latestBlocksCount = 0; // replace with actual data
    const updateList = jest.fn();
    const enablePagination = false;
    const isFetchingNextPage = false;
    const hasNextPage = false;
    const fetchNextPage = jest.fn();

    const { asFragment } = render(
      <BlockListProvider>
        <BlockListWithControls
          blockList={blockList}
          latestBlocksCount={latestBlocksCount}
          updateList={updateList}
          enablePagination={enablePagination}
          isFetchingNextPage={isFetchingNextPage}
          hasNextPage={hasNextPage}
          fetchNextPage={fetchNextPage}
        />
      </BlockListProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
