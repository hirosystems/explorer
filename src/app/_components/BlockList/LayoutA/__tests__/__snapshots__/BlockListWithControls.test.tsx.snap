// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockListWithControls renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-16e8ooo"
  >
    <div
      class="css-1rdddxq"
    >
      <span
        class="chakra-text css-hnwgpl"
      >
        Recent Blocks
      </span>
    </div>
    <div
      class="css-79elbk"
    >
      <div
        class="css-1gdm77d"
      >
        <div
          class="chakra-stack css-1lzfyzp"
        >
          <div
            class="css-gg4vpm"
          >
            <div
              class="chakra-form-control css-1v0vze3"
              role="group"
            >
              <label
                class="chakra-switch css-1tv0hy9"
                data-checked=""
                data-disabled=""
              >
                <input
                  aria-disabled="true"
                  aria-invalid="false"
                  checked=""
                  class="chakra-switch__input"
                  disabled=""
                  id="group-by-btc"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value=""
                />
                <span
                  aria-hidden="true"
                  class="chakra-switch__track css-14qxnv8"
                  data-checked=""
                  data-disabled=""
                >
                  <span
                    class="chakra-switch__thumb css-0"
                    data-checked=""
                    data-disabled=""
                  />
                </span>
              </label>
              <label
                class="chakra-form__label css-2edo7i"
                colo="textCaption"
                for="group-by-btc"
                id="field-:r0:-label"
              >
                Group by Bitcoin block
              </label>
            </div>
          </div>
          <div
            class="chakra-form-control css-guqklo"
            role="group"
          >
            <label
              class="chakra-switch css-1tv0hy9"
            >
              <input
                aria-disabled="false"
                aria-invalid="false"
                class="chakra-switch__input"
                id="live-updates"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-switch__track css-14qxnv8"
              >
                <span
                  class="chakra-switch__thumb css-0"
                />
              </span>
            </label>
            <label
              class="chakra-form__label css-2edo7i"
              colo="textCaption"
              for="live-updates"
              id="field-:r1:-label"
            >
              Live updates
            </label>
          </div>
        </div>
        <div
          class="css-1g6pedm"
          style="transition: opacity 0.7s; opacity: 1;"
        >
          <span
            class="chakra-text css-1eiok6j"
          >
            <span
              class="chakra-text css-1csk9zj"
            >
              0
            </span>
             new Stacks blocks have come in
          </span>
          <span
            class="chakra-text css-1q6cug2"
          >
            <div
              class="css-172hjuw"
            >
              <svg
                class="chakra-icon css-1dgznug"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                version="1.1"
                viewBox="0 0 17 17"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g />
                <path
                  d="M6 8h-6v-6h1v4.109c1.013-3.193 4.036-5.484 7.5-5.484 3.506 0 6.621 2.36 7.574 5.739l-0.963 0.271c-0.832-2.95-3.551-5.011-6.611-5.011-3.226 0.001-6.016 2.276-6.708 5.376h4.208v1zM11 9v1h4.208c-0.693 3.101-3.479 5.375-6.708 5.375-3.062 0-5.78-2.061-6.611-5.011l-0.963 0.271c0.952 3.379 4.067 5.739 7.574 5.739 3.459 0 6.475-2.28 7.5-5.482v4.108h1v-6h-6z"
                />
              </svg>
              <span
                class="chakra-text css-arsoxi"
              >
                Update
              </span>
            </div>
          </span>
        </div>
        <div
          class="chakra-stack css-1dg20yu"
          style="transition: opacity 0.7s; opacity: 1;"
        >
          <div
            class="css-c7wx4b"
          >
            <div
              class="css-g8y6qd"
            >
              <div
                class="css-k008qs"
              >
                <div
                  class="css-1922wes"
                >
                  <svg
                    class="chakra-icon chakra-icon css-9ha1hw"
                    focusable="false"
                    size="2.5"
                    viewBox="0 0 25 25"
                  >
                    <path
                      d="M16.92 7.87828C16.8288 7.72184 16.8418 7.5263 16.946 7.36986L21.2951 0.916845C21.4123 0.734336 21.4253 0.512717 21.3211 0.330207C21.217 0.134662 21.0216 0.0303704 20.8133 0.0303704H19.1205C18.9382 0.0303704 18.756 0.121625 18.6388 0.291098L13.5605 7.85221C13.4303 8.04775 13.2219 8.15204 12.9875 8.15204H12.3495C12.1151 8.15204 11.9068 8.03472 11.7766 7.85221L6.72431 0.278062C6.62014 0.108589 6.42482 0.017334 6.24253 0.017334H4.54976C4.34142 0.017334 4.13308 0.134662 4.04193 0.330207C3.93776 0.525753 3.96381 0.747372 4.06798 0.916845L8.41708 7.3829C8.52125 7.5263 8.53427 7.72184 8.44312 7.87828C8.35197 8.04775 8.19571 8.13901 8.01342 8.13901H1.35956C1.03402 8.13901 0.786621 8.39973 0.786621 8.71261V10.1205C0.786621 10.4464 1.04705 10.6941 1.35956 10.6941H24.0035C24.329 10.6941 24.5765 10.4334 24.5765 10.1205V8.71261C24.5765 8.41277 24.3551 8.17812 24.0686 8.13901C24.0426 8.13901 24.0165 8.13901 23.9905 8.13901H17.3497C17.1674 8.13901 16.9981 8.04775 16.92 7.87828ZM11.7896 16.1825L6.71129 23.7436C6.60712 23.913 6.4118 24.0043 6.2295 24.0043H4.53674C4.3284 24.0043 4.13308 23.887 4.02891 23.7045C3.92474 23.5219 3.93776 23.2873 4.05496 23.1178L8.39103 16.6648C8.4952 16.5084 8.50822 16.3259 8.41708 16.1564C8.32593 16 8.16967 15.8957 7.98737 15.8957H1.35956C1.04705 15.8957 0.786621 15.648 0.786621 15.3221V13.9141C0.786621 13.6013 1.03402 13.3405 1.35956 13.3405H23.9514C23.9514 13.3405 23.9905 13.3405 24.0035 13.3405C24.316 13.3405 24.5765 13.5882 24.5765 13.9141V15.3221C24.5765 15.6349 24.329 15.8957 24.0035 15.8957H17.3627C17.1674 15.8957 17.0111 15.9869 16.933 16.1564C16.8418 16.3259 16.8548 16.5084 16.959 16.6518L21.3081 23.1178C21.4123 23.2873 21.4383 23.5089 21.3342 23.7045C21.23 23.9 21.0347 24.0173 20.8263 24.0173H19.1336C18.9382 24.0173 18.769 23.9261 18.6648 23.7696L13.5865 16.2085C13.4563 16.013 13.248 15.9087 13.0136 15.9087H12.3755C12.1412 15.9087 11.9328 16.026 11.8026 16.2085L11.7896 16.1825Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <a
                  class="chakra-link css-0"
                  href="/block/hash1?chain=mainnet"
                >
                  <span
                    class="chakra-text css-1wk7lue"
                  >
                    #1
                  </span>
                </a>
              </div>
              <div
                class="chakra-stack css-14uiuhs"
              >
                <div
                  class="css-0"
                >
                  has…sh1
                </div>
                 ∙ 
                <div
                  class="css-0"
                >
                  5 txn
                </div>
                 ∙ 
                <div
                  class="css-70qvj9"
                >
                  <span
                    class="chakra-text css-nvxyq8"
                  >
                    in 53946 years
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-8mx47l"
          >
            <div
              class="css-hu3jac"
            >
              <div
                class="css-k008qs"
              >
                <a
                  class="chakra-link css-0"
                  href="/block/hash2?chain=mainnet"
                >
                  <span
                    class="chakra-text css-1wk7lue"
                  >
                    #2
                  </span>
                </a>
              </div>
              <div
                class="chakra-stack css-14uiuhs"
              >
                <div
                  class="css-0"
                >
                  has…sh2
                </div>
                 ∙ 
                <div
                  class="css-0"
                >
                  10 txn
                </div>
                 ∙ 
                <div
                  class="css-70qvj9"
                >
                  <span
                    class="chakra-text css-nvxyq8"
                  >
                    in 53946 years
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-8mx47l"
          >
            <div
              class="css-hu3jac"
            >
              <div
                class="css-k008qs"
              >
                <a
                  class="chakra-link css-0"
                  href="/block/hash3?chain=mainnet"
                >
                  <span
                    class="chakra-text css-1wk7lue"
                  >
                    #3
                  </span>
                </a>
              </div>
              <div
                class="chakra-stack css-14uiuhs"
              >
                <div
                  class="css-0"
                >
                  has…sh3
                </div>
                 ∙ 
                <div
                  class="css-0"
                >
                  15 txn
                </div>
                 ∙ 
                <div
                  class="css-70qvj9"
                >
                  <span
                    class="chakra-text css-nvxyq8"
                  >
                    in 53946 years
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-k0mx8y"
          >
            <div
              class="chakra-stack css-1xwk49h"
            >
              <svg
                class="chakra-icon css-12nniua"
                fill="currentColor"
                focusable="false"
                height="2.5"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="2.5"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14.5 1.5a.5.5 0 0 1 .5.5v4.8a2.5 2.5 0 0 1-2.5 2.5H2.707l3.347 3.346a.5.5 0 0 1-.708.708l-4.2-4.2a.5.5 0 0 1 0-.708l4-4a.5.5 0 1 1 .708.708L2.707 8.3H12.5A1.5 1.5 0 0 0 14 6.8V2a.5.5 0 0 1 .5-.5z"
                  fill-rule="evenodd"
                />
              </svg>
              <svg
                class="chakra-icon chakra-icon css-1t1nt6"
                focusable="false"
                size="18"
                viewBox="0 0 19 19"
              >
                <g
                  clip-path="url(#clip0_48_31)"
                >
                  <path
                    d="M17.7585 11.997C16.5564 16.8184 11.6731 19.7527 6.85107 18.5504C2.03101 17.3483 -0.903273 12.4647 0.299352 7.64351C1.50085 2.82148 6.3842 -0.113082 11.2048 1.08898C16.0266 2.29104 18.9606 7.17523 17.7585 11.997Z"
                    fill="#F7931A"
                  />
                  <path
                    d="M12.9964 8.53766C13.1755 7.3401 12.2637 6.69632 11.0169 6.26685L11.4214 4.6446L10.4339 4.39851L10.0402 5.97801C9.78057 5.91332 9.51394 5.85229 9.249 5.79182L9.64557 4.20191L8.65866 3.95582L8.25394 5.57751C8.03907 5.52857 7.82813 5.48019 7.62338 5.42929L7.6245 5.42422L6.26269 5.08419L6 6.13888C6 6.13888 6.73266 6.30679 6.71719 6.31719C7.11713 6.41704 7.18941 6.68169 7.17732 6.89151L6.71663 8.7396C6.74419 8.74663 6.77991 8.75676 6.81928 8.77251C6.78638 8.76435 6.75122 8.75535 6.71494 8.74663L6.06919 11.3355C6.02025 11.457 5.89622 11.6393 5.61666 11.5701C5.6265 11.5844 4.89891 11.3909 4.89891 11.3909L4.40869 12.5213L5.69372 12.8416C5.93279 12.9015 6.16707 12.9643 6.39769 13.0233L5.98904 14.6641L6.97538 14.9102L7.3801 13.2868C7.64954 13.36 7.9111 13.4275 8.16703 13.491L7.76372 15.1068L8.75119 15.3529L9.15985 13.7152C10.8437 14.0338 12.1099 13.9053 12.6428 12.3823C13.0723 11.1561 12.6215 10.4488 11.7355 9.98751C12.3807 9.83872 12.8667 9.41432 12.9964 8.53766ZM10.7402 11.7014C10.435 12.9277 8.37038 12.2648 7.701 12.0986L8.24325 9.92479C8.91263 10.0918 11.0591 10.4226 10.7402 11.7014ZM11.0456 8.51994C10.7672 9.63538 9.04875 9.06866 8.49132 8.92972L8.98294 6.95816C9.54038 7.0971 11.3356 7.35641 11.0456 8.51994Z"
                    fill="white"
                  />
                </g>
              </svg>
              <a
                class="chakra-text css-0"
                href="https://mempool.space/block/hash4"
                target="_blank"
              >
                <span
                  class="chakra-text css-6xh8zb"
                >
                  #4
                </span>
              </a>
            </div>
            <div
              class="chakra-stack css-28v9tq"
            >
              <div
                class="css-0"
              >
                has…sh4
              </div>
               ∙ 
              <div
                class="css-70qvj9"
              >
                <span
                  class="chakra-text css-nvxyq8"
                >
                  in 53946 years
                </span>
              </div>
            </div>
          </div>
          <div
            class="css-1jixim0"
          >
            <a
              class="chakra-link css-0"
              href="/blocks?chain=mainnet"
            >
              <span
                class="chakra-text css-1cy56wk"
              >
                +25 blocks
                <div
                  class="css-10s7be0"
                >
                  <svg
                    class="chakra-icon css-1nguf4z"
                    fill="currentColor"
                    focusable="false"
                    height="2.5"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 256 256"
                    width="2.5"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M200,64V168a8,8,0,0,1-16,0V83.31L69.66,197.66a8,8,0,0,1-11.32-11.32L172.69,72H88a8,8,0,0,1,0-16H192A8,8,0,0,1,200,64Z"
                    />
                  </svg>
                </div>
              </span>
            </a>
          </div>
        </div>
        <div
          class="css-37urdo"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
